torchvision/models/swin_transformer.py

```diff
@@ -218,8 +231,12 @@ def forward(self, x: Tensor):
         Returns:
             Tensor with same layout as input, i.e. [B, H, W, C]
         """
+        _, H, W, _ = x.shape
+        input_hw = [H, W]
+        # Handle case where the window_size is larger than the input
+        window_size, shift_size = _fix_window_and_shift_size(input_hw, self.window_size, self.shift_size)
```

Comment:
The [original implementation](https://github.com/microsoft/Swin-Transformer/blob/main/models/swin_transformer.py#L192-L195) do this in the constructor. However in the original implementation they require to have `input_resolution` in the constructor which we dont have here. Therefore I do this in the forward so I can access the input size.

Reply:
The issue here is you modify the `self.window_size` and `self.shift_size` methods in-place, leaving side-effects. Is this intentional? If not we should copy them. If yes, it might be worth reviewing the logic, because as far as I understand once the values are modified you currently have no way to recover the original values.

Reply:
Thanks for noticing this @datumbox ! Yes, I think we shouldn't modify `self.window_size` and this is currently not intentional. I will change the function `_fix_window_and_shift_size` so that it does not modify the class variable.